packages/aws-cdk/lib/api/cxapp/exec.ts

```diff
@@ -78,10 +78,12 @@ export async function execProgram(aws: SdkProvider, config: Configuration): Prom
     try {
       return new cxapi.CloudAssembly(appDir);
     } catch (error) {
-      if (error.message.includes(cxschema.VERSION_MISMATCH)) {
+      if (error.versionMismatch && error.manifestVersion) {
         // this means the CLI version is too old.
         // we instruct the user to upgrade.
-        throw new Error(`${error.message}.\nPlease upgrade your CLI in order to interact with this app.`);
+
+        throw new Error('The CLI version you are using does not support your application version. Please upgrade the CLI to the latest version.'
+        + `\n(Your application requires a CLI that supports a cloud assembly of version '${error.manifestVersion}' or above.)`);
```

Comment:
Let's not be smart about this... Just append the original error message like this (also made some changes to the phrasing):

```suggestion
        throw new Error('This CDK CLI is not compatible with the CDK library used by your application. Please upgrade the CLI to the latest version (${error.message})`);
```

This will result in a decent error:

```
This CDK CLI is not compatible with the CDK library used by your application. 
Please upgrade the CLI to the latest version (Cloud assembly schema version 
mismatch: Maximum schema version supported is 1.32.0, but found 3.0.0)
```

Reply:
👍